blueflood-http/src/main/java/com/rackspacecloud/blueflood/utils/DateTimeParser.java

```diff
@@ -0,0 +1,187 @@
+package com.rackspacecloud.blueflood.utils;
+
+import org.apache.commons.lang.StringUtils;
+import org.joda.time.DateTime;
+import org.joda.time.format.DateTimeFormat;
+
+import java.util.Arrays;
+import java.util.List;
+import java.util.regex.Matcher;
+import java.util.regex.Pattern;
+
+public class DateTimeParser {
```

Comment:
While I appreciate the comprehensiveness of the `DateTimeParser` class, I am not sure whether we really need do much of flexibility with supplying the dates...I am going to look into graphite api for a guideline, but if anyone has any strong opinions around the necessity of this class, let me know.